### PR TITLE
CRV LP Strategy abstraction

### PR DESCRIPTION
**Pull request by pajicf**
## Checklist (Mandatory):

- [x] I have named the PR correctly
- [x] I have tested new code locally
- [x] I have done a smoke test of the larger feature locally
- [x] I have added at least one reviewer
- [x] I have assigned the PR to myself
- [x] I have merged develop (parent) branch into my feature branch and resolved all conflicts

## Additional notes

```
What does this implement/fix? Explain your changes.
- Refactor to support multiple pools in the same project
- AAVE pool for CRV
Are there any edge cases that need checking?
/
Are there any specific test instructions?
/
```

### CODE REVIEW

File: curve-lp/contracts/StrategyCurveAave.sol [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            crvBalance = crv.balanceOf(address(this));
+        }
+
+        // sell our WAVAX and CRV if we have any
+        if (wavaxBalance > 0 || crvBalance > 0) {
+            _sell(wavaxBalance, crvBalance);
+        }
+
+        // deposit our balance to Curve if we have any
+        if (optimal == 0) {
```
Comment (joshualyguessennd) [inline curve-lp/contracts/StrategyCurveAave.sol:85]: is it possible to set the same logic as minichef, and remove the ifs as Val suggest ? in case we switch optimal without harvesting , we deposit the dust found if available to the pool

**Comment by joshualyguessennd** [comment]
the code  🚀

File: curve-lp/contracts/StrategyCurveAave.sol (hunk 1 above)
Comment (pajicf) [reply curve-lp/contracts/StrategyCurveAave.sol:85]: Rationale here is that we save potentially unnecessary `balanceOf` calls and thereby save ~5k gas per tx.
I don't think there should be any dust ever since selling to targetToken and depositing is done in the same tx, during prepareReturn

**Comment by valo** [comment]
LGTM

**Review by valo — APPROVED** [review]
